### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by kostya-konev**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,35 @@
...
+  const newArr = [];
+  const newObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const prop in action.extraData) {
+          newObj[prop] = action.extraData[prop];
+        }
+        newArr.push({ ...newObj });
```
Comment (marina-tilniak) [inline src/transformStateWithClones.js:19]: use Object.assign here

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,35 @@
...
+        break;
+
+      case 'clear':
+        for (const prop in newObj) {
+          delete newObj[prop];
+        }
+        newArr.push({ ...newObj });
+        break;
+    }
+  }
```
Comment (marina-tilniak) [inline src/transformStateWithClones.js:36]:
```suggestion
    }
    newArr.push({  ...newObj });
  }
```

Comment (marina-tilniak) [reply src/transformStateWithClones.js:36]: to do push only one time here

**Review by marina-tilniak — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js (hunk 1 above)
Comment (kostya-konev) [reply src/transformStateWithClones.js:19]: Why?

File: src/transformStateWithClones.js (hunk 2 above)
Comment (kostya-konev) [reply src/transformStateWithClones.js:36]: Why?

**Review by ArtemTopolskyi — APPROVED** [review]

**Pull request closed without merging**